pootle/apps/pootle_format/utils.py

```diff
@@ -63,3 +76,99 @@ def valid_extensions(self):
             OrderedDict.fromkeys(
                 self.filetype_extensions
                 + self.template_extensions))
+
+    def add_filetype(self, filetype):
+        """Adds a filetype to a Project
+        """
+        if filetype not in self.filetypes:
+            log.info(
+                "Adding filetype '%s' to project '%s'",
+                filetype, self.project)
+            self.project.filetypes.add(filetype)
+
+    def set_store_filetype(self, store, filetype):
+        """Sets a Store to given filetype
+
+        If `from_filetype` is given, only updates if current filetype matches
+        """
+        if filetype not in self.filetypes:
+            raise self.unrecognized_filetype(filetype)
+        log.info(
+            "Setting filetype '%s' to store '%s'",
+            filetype, store.pootle_path)
+        store.filetype = filetype
+        # update the extension if required
+        extension = (
+            store.is_template
+            and str(filetype.template_extension)
+            or str(filetype.extension))
+        root_name = os.path.splitext(store.name)[0]
+        new_name = (
+            root_name.endswith(".%s" % extension)
+            and root_name
+            or ("%s.%s"
+                % (root_name, extension)))
+        if store.name != new_name:
+            store.name = new_name
+        store.save()
+
+    def _tp_path_regex(self, tp, matching):
+        """Creates a regex from
+
+        /tp/path/$glob_converted_to_regex.($valid_extensions)$
+        """
+        extensions = (
+            (tp == self.project.get_template_translationproject())
+            and self.template_extensions
+            or self.filetype_extensions)
+        return (
+            r"^/%s/%s/%s.%s$"
```

Comment:
feedback from @ta2-1 - this fails

```
>>> re.match("foo.(bar|baz)", "fooobar")
<_sre.SRE_Match object at 0x7fe684e28648>
```

so conclusion is to add just a single slash!